nncf/quantization/algorithms/weight_compression/mixed_precision.py

```diff
@@ -0,0 +1,162 @@
+        return weight_score
+
+    def _calc_scores(self):
+        scores = []
+        for weight_param in track(self._weight_params, description="Searching for Mixed-Precision Configuration"):
+            scores.append(self._calc_score_per_node(weight_param))
+        return scores
+
+
+class DataBasedCriterion(DataFreeCriterion):
+    @staticmethod
+    @abstractmethod
+    def _calc_activation_score(activations: np.ndarray):
+        """
+        activation shape = [seq_length, hidden_dim]
+        """
+        pass
+
+    def _calc_score_per_node(self, weight_param: WeightNodeParams):
+        weight_score = self._calc_weight_score(weight_param)
+        activation_score = self._calc_activation_score(self._activations[weight_param.node_name])
+        return weight_score * activation_score
+
+
+@MIXED_PRECISION_CRITERIA.register(MixedPrecisionMode.HAWQ_IN)
+class HAWQCriterion(DataBasedCriterion):
+    @staticmethod
+    def _calc_activation_score(activations: np.ndarray):
+        htrace = 0
+        nsamples = len(activations)
+        for inp in activations:
+            # NOTE: average trace?? divide by number of diagonal elements
+            htrace += np.sum(np.multiply(inp, inp))
+            # normalize by sequence_length - the same for all activations
+            # normalize by hidden dimension
+            htrace /= inp.size
+        htrace *= 2 / nsamples
+        return htrace
+
+    @staticmethod
+    def _calc_weight_score(weight_param: WeightNodeParams):
+        weight = get_const_value(weight_param.weight_node)
+        backup_config = weight_param.compression_config
+        reduction_axis = weight_param.reduction_axis
+
+        orig_shape = weight.shape
+        compressed_weights, scale, zero_point = _do_integer_quantization(weight, reduction_axis, backup_config)
+        decompressed_weight = compressed_weights.astype(dtype=scale.dtype)
+        decompressed_weight = (compressed_weights - zero_point) * scale
+        decompressed_weight = decompressed_weight.reshape(orig_shape)
+        return linalg.norm(decompressed_weight - weight, ord="fro")
+
+
+@MIXED_PRECISION_CRITERIA.register(MixedPrecisionMode.MEAN_VAR)
+class MeanVarianceCriterion(DataBasedCriterion):
+    @staticmethod
+    def _calc_activation_score(activations: np.ndarray):
+        return float(np.mean([np.mean(np.var(inp, axis=0)) for inp in activations]))
+
+
+@MIXED_PRECISION_CRITERIA.register(MixedPrecisionMode.MAX_VAR)
+class MaxVarianceCriterion(DataBasedCriterion):
+    @staticmethod
+    def _calc_activation_score(activations: np.ndarray):
+        return float(np.mean([np.max(np.var(inp, axis=0)) for inp in activations]))
+
+
+@MIXED_PRECISION_CRITERIA.register(MixedPrecisionMode.MEAN_MAX)
+class MeanMaxCriterion(DataBasedCriterion):
+    @staticmethod
+    def _calc_activation_score(activations: np.ndarray):
+        return float(np.mean([np.mean(np.max(inp, axis=0)) for inp in activations]))
```

Comment:
Why is it not for absolute values?

Reply:
good catch! should be max, will correct it